freqtrade/exchange/exchange.py

```diff
@@ -887,14 +887,20 @@ async def _async_get_trade_history(self, pair: str,
         Async wrapper handling downloading trades using either time or id based methods.
         """
 
+        logger.debug(f"_async_get_trade_history(), pair: {pair}, "
+                     f"since: {since}, until: {until}, from_id: {from_id}")
+
+        if not until:
+            exchange_msec = ccxt.Exchange.milliseconds()
+            logger.debug(f"Exchange milliseconds: {exchange_msec}")
+            until = exchange_msec
+
```

Comment:
You are aware what Exchange.milliseconds() does - it's just giving you the time of your system.... 
i don't see why we would need to log this ... ?

Reply:
* I did not change the logic here -- if until have used ccxt.Exchange.milliseconds(), it still uses it
* This solves the following problem: when I run `download-data --dl-trades -v`, with debug verbosity, I can see the timestamps of the chunks downloaded:
```
2020-06-02 18:14:26,665 - freqtrade.data.history.history_utils - INFO - Downloading trades for pair ADA/EUR.
2020-06-02 18:14:26,665 - freqtrade.data.history.history_utils - DEBUG - Current Start: None
2020-06-02 18:14:26,665 - freqtrade.data.history.history_utils - DEBUG - Current End: None
2020-06-02 18:14:26,665 - freqtrade.data.history.history_utils - INFO - Current Amount of trades: 0
2020-06-02 18:14:26,665 - freqtrade.exchange.exchange - DEBUG - Fetching trades for pair ADA/EUR, since 1582416000000 (2020-02-23T00:00:00+00:00) ...
2020-06-02 18:14:27,299 - freqtrade.exchange.exchange - DEBUG - Fetching trades for pair ADA/EUR, params: {'since': '1582472451120542812'} 
2020-06-02 18:14:27,935 - freqtrade.exchange.exchange - DEBUG - Fetching trades for pair ADA/EUR, params: {'since': '1582540089278740091'} 
2020-06-02 18:14:28,533 - freqtrade.exchange.exchange - DEBUG - Fetching trades for pair ADA/EUR, params: {'since': '1582571623919118694'} 
2020-06-02 18:14:29,130 - freqtrade.exchange.exchange - DEBUG - Fetching trades for pair ADA/EUR, params: {'since': '1582627937743271731'} 
2020-06-02 18:14:29,719 - freqtrade.exchange.exchange - DEBUG - Fetching trades for pair ADA/EUR, params: {'since': '1582663237842671533'} 
2020-06-02 18:14:30,797 - freqtrade.exchange.exchange - DEBUG - Fetching trades for pair ADA/EUR, params: {'since': '1582701444635332773'} 
2020-06-02 18:14:31,511 - freqtrade.exchange.exchange - DEBUG - Fetching trades for pair ADA/EUR, params: {'since': '1582722112264136533'} 
2020-06-02 18:14:32,238 - freqtrade.exchange.exchange - DEBUG - Fetching trades for pair ADA/EUR, params: {'since': '1582736123676981905'} 
2020-06-02 18:14:32,873 - freqtrade.exchange.exchange - DEBUG - Fetching trades for pair ADA/EUR, params: {'since': '1582753848031010356'} 
```
But I have no idea what is the final timestamp for this row of downloads, how many chunks it will download and how long time should I expect it to take with my `rateLimit`... This PR adds printing of the `until` timestamp, to see if it only downloads some final chunks or starts for the whole set for `--days` (if there is no data for the pair)...

Reply:
Now it prints the final timestamp (used as 'until') and helps me see in this case that I can go smoke a cigarette:
```
2020-06-02 18:23:36,544 - freqtrade.data.history.history_utils - INFO - Downloading trades for pair ADA/EUR.
2020-06-02 18:23:36,544 - freqtrade.data.history.history_utils - DEBUG - Current Start: None
2020-06-02 18:23:36,544 - freqtrade.data.history.history_utils - DEBUG - Current End: None
2020-06-02 18:23:36,545 - freqtrade.data.history.history_utils - INFO - Current Amount of trades: 0
2020-06-02 18:23:36,545 - freqtrade.exchange.exchange - DEBUG - _async_get_trade_history(), pair: ADA/EUR, since: 1582416000000, until: None, from_id: None
2020-06-02 18:23:36,545 - freqtrade.exchange.exchange - DEBUG - Exchange milliseconds: 1591111416545
2020-06-02 18:23:36,545 - freqtrade.exchange.exchange - DEBUG - Fetching trades for pair ADA/EUR, since 1582416000000 (2020-02-23T00:00:00+00:00) ...
2020-06-02 18:23:37,836 - freqtrade.exchange.exchange - DEBUG - Fetching trades for pair ADA/EUR, params: {'since': '1582472451120542812'} 
2020-06-02 18:23:38,398 - freqtrade.exchange.exchange - DEBUG - Fetching trades for pair ADA/EUR, params: {'since': '1582540089278740091'} 
2020-06-02 18:23:38,877 - freqtrade.exchange.exchange - DEBUG - Fetching trades for pair ADA/EUR, params: {'since': '1582571623919118694'} 
2020-06-02 18:23:40,138 - freqtrade.exchange.exchange - DEBUG - Fetching trades for pair ADA/EUR, params: {'since': '1582627937743271731'} 
2020-06-02 18:23:41,811 - freqtrade.exchange.exchange - DEBUG - Fetching trades for pair ADA/EUR, params: {'since': '1582663237842671533'} 
2020-06-02 18:23:42,519 - freqtrade.exchange.exchange - DEBUG - Fetching trades for pair ADA/EUR, params: {'since': '1582701444635332773'} 
...
```
and here it only downloads small amount of latest data:
```
2020-06-02 18:26:11,947 - freqtrade.data.history.history_utils - INFO - Downloading trades for pair XTZ/EUR.
2020-06-02 18:26:11,947 - freqtrade.misc - DEBUG - Loading historical data from file /home/user/freqtrade-wrk/github-hroff-1902/freqtrade/user_data/data/kraken/XTZ_EUR-trades.json.gz
2020-06-02 18:26:13,043 - freqtrade.data.history.history_utils - INFO - Using last trade date -5s - Downloading trades for XTZ/EUR since: 2020-06-02T07:10:14.
2020-06-02 18:26:13,043 - freqtrade.data.history.history_utils - DEBUG - Current Start: 2020-02-21T03:00:09
2020-06-02 18:26:13,043 - freqtrade.data.history.history_utils - DEBUG - Current End: 2020-06-02T07:10:19
2020-06-02 18:26:13,043 - freqtrade.data.history.history_utils - INFO - Current Amount of trades: 382974
2020-06-02 18:26:13,043 - freqtrade.exchange.exchange - DEBUG - _async_get_trade_history(), pair: XTZ/EUR, since: 1591071014343, until: None, from_id: None
2020-06-02 18:26:13,043 - freqtrade.exchange.exchange - DEBUG - Exchange milliseconds: 1591111573043
2020-06-02 18:26:13,044 - freqtrade.exchange.exchange - DEBUG - Fetching trades for pair XTZ/EUR, since 1591071014343 (2020-06-02T04:10:14+00:00) ...
2020-06-02 18:26:13,695 - freqtrade.exchange.exchange - DEBUG - Fetching trades for pair XTZ/EUR, params: {'since': '1591086787149656824'} 
2020-06-02 18:26:14,173 - freqtrade.exchange.exchange - DEBUG - Fetching trades for pair XTZ/EUR, params: {'since': '1591100996841328283'} 
2020-06-02 18:26:15,261 - freqtrade.exchange.exchange - DEBUG - Fetching trades for pair XTZ/EUR, params: {'since': '1591109412206605753'} 
2020-06-02 18:26:15,785 - freqtrade.exchange.exchange - DEBUG - Fetching trades for pair XTZ/EUR, params: {'since': '1591111562820291652'} 
```